code/game/objects/effects/traps.dm

```diff
@@ -438,31 +439,38 @@ Add those other swinging traps you mentioned above!
 	return
 
 /obj/effect/trap/pop_up/attackby(var/obj/item/W, var/mob/user)
+	if(istype(W, /obj/item/weldingtool))
+		var/obj/item/weldingtool/WT = W
+
+		if(WT.remove_fuel(0, user))
+			if(health < maxhealth)
+				to_chat(user, "<span class='notice'>You begin repairing \the [src.name] with \the [WT].</span>")
+			if(do_after(user, 20, src))
+				health = maxhealth
+				broken = FALSE
+			playsound(src.loc, 'sound/items/Welder.ogg', 100, 1)
+
+	if(broken)
+		return
 	if((health <= 0))
 		Break()
+		src.visible_message("<span class='danger'>\The [src] breaks! It was a trap!</span>")
 		return
 	if(W.attack_verb.len)
 		src.visible_message("<span class='danger'>\The [src] has been [pick(W.attack_verb)] with \the [W][(user ? " by [user]." : ".")]</span>")
 	else
 		src.visible_message("<span class='danger'>\The [src] has been attacked with \the [W][(user ? " by [user]." : ".")]</span>")
 	var/damage = W.force / 4.0
 
-	if(istype(W, /obj/item/weldingtool))
-		var/obj/item/weldingtool/WT = W
 
-		if(WT.remove_fuel(0, user))
-			if(health < maxhealth)
-				to_chat(user, "<span class='notice'>You begin repairing \the [src.name] with \the [WT].</span>")
-			if(do_after(user, 20, src))
-				health = maxhealth
-			playsound(src.loc, 'sound/items/Welder.ogg', 100, 1)
 
 	src.health -= damage
 	healthcheck()
 
 /obj/effect/trap/pop_up/proc/healthcheck()
 	if((health <= 0))
 		Break()
+		src.visible_message("<span class='danger'>\The [src] breaks! It was a trap!</span>")
```

Comment:
Won't this be double posted since this is called by `/obj/effect/trap/pop_up/attackby`?